incrementalbuild/src/main/java/io/takari/incrementalbuild/maven/internal/MavenBuildContextFinalizer.java

```diff
@@ -105,11 +105,23 @@ protected void failBuild(final Map<Object, Collection<Message>> messages)
         }
       }
     }
-    if (errorCount > 0) {
+
+    if (errorCount > 0 && isFailOnErrorPresent(contexts)) {
       throw new MojoExecutionException(errorCount + " error(s) encountered:\n" + errors.toString());
     }
   }
 
+  private boolean isFailOnErrorPresent(List<AbstractBuildContext> contexts){
+    boolean failOnError = true;
+    for (AbstractBuildContext context : contexts) {
+      if(!context.isFailOnError()){
+        failOnError = false;
```

Comment:
If one context wants to fail the build, the build should fail in case of errors. (I think you have it the other way around, i.e. even if one context does not want to fail, the build won't fail).

Reply:
the problem of your suggestion is that the default value is `true` what would imply that we'd never get the failOnError working with multiple contexts.
to address your point here we'd need to change data type from boolean to an enum with 3 states: NONE (default), TRUE, FALSE.
So the options are:
 - keep as proposed (what you didn't like much)
 - use enum
what you prefer?

Reply:
Multiple context is somewhat obscure feature that allows the same mojo do multiple things, for example produce aggregate and individual outputs. I think this code can throw `IllagalStateException` if different contexts disagree what to do in case of errors. Alternatively, change the code to treat errors from different contexts differently. Either way, letting one context decide how to treat errors in another context would be wrong.